### PR TITLE
Improve context based healthchecks

### PR DESCRIPTION
**Pull request by HerrDerb**

### CODE REVIEW

File: testsuite/experimental/src/test/java/io/smallrye/health/deployment/ContextHealthCheckRegistrar.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import jakarta.inject.Inject;
+
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+
+import io.smallrye.health.api.HealthRegistry;
+import io.smallrye.health.api.Wellness;
+
+@ApplicationScoped
+public class ContextHealthCheckRegistrar {
```
Comment (xstefank) [inline testsuite/experimental/src/test/java/io/smallrye/health/deployment/ContextHealthCheckRegistrar.java:15]: This class is not used by the test.

**Review by xstefank — CHANGES_REQUESTED** [review]
Looks good, just a few small changes needed.

**Comment by xstefank** [comment]
@HerrDerb, if you could also squash the commits, it would be great. Thanks!

File: implementation/src/main/java/io/smallrye/health/registry/HealthRegistryImpl.java [hunk 2]
```diff
@@ -16,31 +16,35 @@
...
-        return register(id, asyncHealthCheckFactory.callAsync(asyncHealthCheck));
+        return register(id, asyncHealthCheck, asyncChecks);
     }
 
-    private HealthRegistry register(String id, Uni<HealthCheckResponse> healthCheckResponseUni) {
-        checks.put(id, healthCheckResponseUni);
+    private <T> HealthRegistry register(String id, T check, Map<String, T> checks) {
+        var alreadRegisteredCheck = checks.putIfAbsent(id, check);
+        if (Objects.nonNull(alreadRegisteredCheck)) {
+            throw new IllegalStateException(String.format("Check for ID '%s' already registerds", id));
```
[the remark this answers is not in the document]
Comment (HerrDerb) [reply implementation/src/main/java/io/smallrye/health/registry/HealthRegistryImpl.java:39]: No this change actually has nothing to do with my usecase. I'll revert that.

**Review by xstefank — APPROVED** [review]

**Comment by xstefank** [comment]
Thanks @HerrDerb

**Pull request merged**